lib/timeout.rb

```diff
@@ -164,6 +164,7 @@ def self.ensure_timeout_thread_created
   # Timeout</tt> into your classes so they have a #timeout method, as well as
   # a module method, so you can call it directly as Timeout.timeout().
   def timeout(sec, klass = nil, message = nil, &block)   #:yield: +sec+
+    raise ArgumentError, "Timeout sec must be a positive number" if sec.is_a?(Numeric) && sec < 0
```

Comment:
didn't raised on non-numeric value, as wanted to keep the compatibility with nil

Reply:
Ah I forgot that the docs explicitly mention nil, and missed that the line below tests for `nil`.
So you are right, we should keep to accept nil, by checking `sec == nil` like the line below.